src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.Unix.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Buffers;
+using System.IO;
+using System.Reflection;
+
+namespace System.Runtime.Loader
+{
+    public partial class AssemblyLoadContext
+    {
+        // Find satellite path using case insensitive culture name
+        internal static unsafe string? FindCaseInsensitiveSatellitePath(string parentDirectory, string cultureName, string assembly)
+        {
+            int bufferSize = Interop.Sys.GetReadDirRBufferSize();
+            byte[]? dirBuffer = null;
+            try
+            {
+                dirBuffer = ArrayPool<byte>.Shared.Rent(bufferSize);
+
+                fixed (byte* dirBufferPtr = dirBuffer)
+                {
+                    IntPtr dirHandle = Interop.Sys.OpenDir(parentDirectory);
+                    if (dirHandle != IntPtr.Zero)
+                    {
+                        try
+                        {
+                            Interop.Sys.DirectoryEntry dirent;
+                            while (Interop.Sys.ReadDirR(dirHandle, dirBufferPtr, bufferSize, out dirent) == 0)
+                            {
+                                if (dirent.InodeType != Interop.Sys.NodeType.DT_DIR)
+                                    continue;
+
+                                Span<char> nameBuffer = stackalloc char[Interop.Sys.DirectoryEntry.NameBufferSize];
+                                ReadOnlySpan<char> entryName = dirent.GetName(nameBuffer);
+
+                                if (cultureName.Length != entryName.Length)
+                                    continue;
+
+                                string entryNameString = entryName.ToString();
```

Comment:
>string entryNameString = entryName.ToString(); [](start = 32, length = 46)

I think there is a way to avoid this allocation. maybe you can get the Span of cultureName and then compare the spans instead of comparing strings.